packages/react-components/react-shared-contexts/src/CustomStyleHooksContext/CustomStyleHooksContext.ts

```diff
@@ -0,0 +1,182 @@
+/* eslint-disable */
+
+import * as React from 'react';
+
+// The list of hooks is built by scraping the exports from react-components/src/index
+export type CustomStyleHooksContextValue = {
+  useAccordionHeaderStyles_unstable: (state: unknown) => void;
```

Comment:
Also, in order to differentiate them more clearly from the built-in style hooks, perhaps these names should include "CustomStyles" e.g.

```suggestion
  useAccordionHeaderCustomStyles: (state: unknown) => void;
```

Reply:
I went back and forth on this a bit.  The custom hooks context is an unstable feature like the overrides context feature.  The hooks inside it are part of the unstable hooks composition model.  If the enclosing context indicate unstable so that the hooks don't need _unstable, then I think the enclosing context would also indicate they are custom styles so wouldn't need CustomStyles in the name.  

I wanted to make it easy for us to keep it in sync with the hooks exported from react-components, so I left the hooks names the same.  It reads well within the component as you get the customstylehooks context and then call the same style hook name you have previously called for the default styling.  Custom in both places felt like too much typing.

I hope that we'll make the hooks composition stable and then we can remove _unstable from the hook methods.  I don't know if we ever want the custom styles context to be stable though.  (I add the _unstable to the custom styles stuff in the exports BTW).